sdk/search/Azure.Search.Documents/tests/Batching/BatchingTests.cs

```diff
@@ -1031,6 +1031,65 @@ public async Task Behavior_BatchSize()
             Assert.AreEqual(5, client.Submissions.Count);
         }
 
+        [Test]
+        public async Task Behavior_SplitBatchByDocumentKey()
```

Comment:
This seems to be testing existing functionality. Given the PR is to make sure only unique actions are batched, I would expect to see a test that has duplicate (even varying only by case - the full gamut of coverage) document keys.

Reply:
Here's the idea behind the test -
1. In line 1051, we set the key value of the 4th document to be the same as the 1st document.
2. Due to that, we expect the batch to split as {1,2,3}, {4,5}, even though the allowed batch size is much larger.
3. I am verifying that by checking the events of actions sent to the publisher and actions submitted by the publisher, which are at document level.
4. Due to the batch split, we assert that the order of operations would be {SSSCCCSSCC} and not the earlier {SSSSSCCCCC}. This is caused by the change in the batching behavior in this PR.

Reply:
Rather than just a number, I'd use a letter as well and make sure it stays case-sensitive.